packages/plugin-hardhat/src/utils/deploy.ts

```diff
@@ -9,7 +11,24 @@ export async function deploy(
   factory: ContractFactory,
   ...args: unknown[]
 ): Promise<Required<Deployment & DeployTransaction>> {
-  const { address, deployTransaction } = await factory.deploy(...args);
+  const contractInstance = await factory.deploy(...args);
+  const deployTransaction = contractInstance.deployTransaction;
+
+  const address: string = getContractAddress({
+    from: await factory.signer.getAddress(),
```

Comment:
We should already have `deployTransaction.from`. In fact, we should be able to just do `getContractAddress(deployTransaction)`.

Reply:
The problem is `deployTransaction.from` returns the 0x0 address, as described in https://forum.openzeppelin.com/t/error-no-contract-at-address-0x-removed-from-manifest/20609/11